RIOT-OS/RIOT
# buildsystem/pkg: expand packages from USEPKG early as path into PKG_PATHS

### Contribution description
This pr it a split from pr #17211 as requested by @fjmolinas.

It changes the way packages are handled. The updated rules now don't explicitly use `RIOTPKG` but instead the new `PKG_PATHS` variable. This new variable holds the path to the packages that were found in `RIOTPKG` and were requested by `USEPKG`. This makes the Makefile rules independent from the actual location of the package. The intention is (with another pr) to allow to seach in additional paths (external to riot) for packages.

### Testing procedure
I've ran `make static-test` with no errors (don't know what exactly is tested though).
I've also compiled the *hello-world* example with `make BOARD=esp8266-esp-12x` or the *suit_update* example.
Regarding a specific test case, I don't think one is needed. As I only change how something is done and don't add anything new, existing tests should cover this pr. CI will tell us :)

### Issues/PRs references
pr #17211
